### PR TITLE
Derive schema from example for attributes/inputs/exits

### PR DESCRIPTION
**Pull request by danielsharvey**
Derive Swagger/OpenAPI schema from example value.

Specifically:
1. The Sails model attribute type 'json' may be best represented in Swagger/OpenAPI as the type 'object' or an array of elements. Let's attempt to determine this from the attribute property 'example'.
2. Actions2 inputs do not necessarily specify a type but may provide example.
3. Actions2 outputs may include `outputExample` but do not specify a type - use this method to derive a schema definition.
4. Add support for request body (in: 'body') in actions2 inputs.

### CODE REVIEW

**Comment by danielsharvey** [comment]
@theoomoregbee This PR now updated with handling of `in: 'body'` (the final part from #66) so should be ready for review and merge.

Note: #82 Should be merged first.

**Comment by theoomoregbee** [comment]
@danielsharvey GitHub switched branch and lost direct history with the new branch. Rebasing should fix the conflicts

**Comment by danielsharvey** [comment]
> @danielsharvey GitHub switched branch and lost direct history with the new branch(master). Rebasing should fix the conflicts

Done

File: api/controllers/subdir/actions2.js [hunk 1]
```diff
@@ -44,22 +49,30 @@ module.exports = {
...
-      outputExample: 'Some dynamic message like this.'
+      description: 'A successful result',
+    },
+    alternateSuccess: {
+      description: 'Alternate success',
+      outputExample: 'Some dynamic message like this.',
+    },
+    alternateSuccess2: {
+      description: 'Alternate success (2)',
+      outputExample: 93.45,
```
Comment (theoomoregbee) [inline api/controllers/subdir/actions2.js:60]: I know this is not related to this PR, but kinda weird that machine as action paradigm uses `example` for input and `outputExample` for exits. Maybe there's a reason for not being consistent and use `example`

**Comment by theoomoregbee** [comment]
Because of the request body attached to get here

Screenshot 2020-06-22 at 22 55 26

**Comment by danielsharvey** [comment]
@theoomoregbee The validator I used did not pick this up! Obvious mistake. Looking.

File: api/controllers/subdir/actions2.js (hunk 1 above)
Comment (danielsharvey) [reply api/controllers/subdir/actions2.js:60]: Yes, there is some inconsistency.

**Comment by danielsharvey** [comment]
@theoomoregbee Added check and warning to ensure verb valid.

File: lib/generators.ts [hunk 2]
```diff
@@ -432,6 +432,11 @@ export const generatePaths = (routes: SwaggerRouteInfo[], templates: BlueprintAc
 
           if(_in === 'body') {
 
+            if(!['put', 'post', 'patch'].includes(route.verb)) {
```
Comment (theoomoregbee) [inline lib/generators.ts:435]: 👍

**Comment by theoomoregbee** [comment]
kinda curious why this didn't create a release, looking at it now

**Pull request merged**